routing/missioncontrol_store.go

```diff
@@ -76,6 +94,10 @@ func newMissionControlStore(db kvdb.Backend, maxRecords int) (*missionControlSto
 
 // clear removes all results from the db.
 func (b *missionControlStore) clear() error {
+	b.queueMx.Lock()
+	defer b.queueMx.Unlock()
+	b.queue = list.New()
```

Comment:
is it ok if we clear the queue here then the delete call fails?

Reply:
Good point, fixed.